src/snowflake/connector/compute_chunk_size.py

```diff
@@ -0,0 +1,69 @@
+#
+# Copyright (c) 2012-2023 Snowflake Computing Inc. All rights reserved.
+#
+
+from __future__ import annotations
+
+import logging
+import math
+
+logger = logging.getLogger(__name__)
+
+constants = {
+    "CURRENT_CHUNK_SIZE": 8388608,
+    "MAX_OBJECT_SIZE": 5497558138880,
+    "MAX_PART_SIZE": 5368709120,
+    "MIN_PART_SIZE": 5242880,
+    "MAX_PARTS": 10000,
+}
+
+
+class ChunkSizeCalculator:
+    def __init__(
+        self,
+        current_chunk_size: int = constants["CURRENT_CHUNK_SIZE"],
+        max_object_size: int = constants["MAX_OBJECT_SIZE"],
+        max_part_size: int = constants["MAX_PART_SIZE"],
+        min_part_size: int = constants["MIN_PART_SIZE"],
+        max_parts: int = constants["MAX_PARTS"],
```

Comment:
can those parameter be of type float?
I saw that in the tests there is a float `sample_file_size_5tb = 4.9 * 1024 * 1024 * 1024 * 1024`

Reply:
Since those are declared by us, I believe they can be of type int, however, I think `file_size` indeed should be of type float in both places: 
```python
    def compute_chunk_size(self, file_size: float) -> int:
``` 
and
```python
    def _check_max_parts(self, current_chunk_size: int, file_size: float) -> int:
``` 
Or should all of them be of type float?

Reply:
yes, in this case we'd better make `sample_file_size_5tb` an integer

Reply:
Done